components/detailsCard/TextareaWithReadMore.tsx

```diff
@@ -0,0 +1,64 @@
+import { Box, Text, Button } from "@chakra-ui/react";
+import { useState } from "react";
+
+export interface TextAreaWithReadMoreProps {
+  orderPolicyText: string;
+  readMoreText: string;
+  readLessText: string;
+}
+
+const TextAreaWithReadMore: React.FC<TextAreaWithReadMoreProps> = (props) => {
+  const [expanded, setExpanded] = useState(false);
+
+  const toggleExpanded = () => {
+    setExpanded(!expanded);
+  };
+
+  const truncatedText = expanded
+    ? props.orderPolicyText
+    : props.orderPolicyText.slice(0, 190) + "...";
+
+  return (
+    <Box padding={"0 15px"} fontSize={"13px"}>
+      <Text
+        textOverflow="ellipsis"
+        whiteSpace="pre-wrap"
+        overflowWrap="break-word"
+        noOfLines={expanded ? undefined : 5}
+        sx={{
+          display: "-webkit-box",
+          WebkitBoxOrient: "vertical",
+          WebkitLineClamp: expanded ? "unset" : 5,
+        }}
+      >
+        {truncatedText}
+        {!expanded && (
```

Comment:
@kundanEminds Since only one condition will be true at a time, use of a ternary operator ( a ? b : c) will be a better fit here. Let me know your thoughts.

Reply:
@ankitShogun  I updated this condition with ternary operator Please check